src/libfsm/fsm.c

```diff
@@ -19,7 +19,57 @@
 #define ctassert(pred) \
 	switch (0) { case 0: case (pred):; }
 
-static void
+void
+f_free(const struct fsm *fsm, void *p)
+{
+	const struct fsm_allocator *a;
+
+	assert(fsm != NULL);
+	assert(fsm->opt != NULL);
```

Comment:
And `fsm->opt->allocator != NULL`

Reply:
The following lines cover the NULL scenario for `fsm->opt->allocator` - it applies the libc free if applicable.